viz_scripts/scaffolding.py

```diff
@@ -197,26 +235,34 @@ def load_viz_notebook_sensor_inference_data(year, month, program, include_test_u
 
     return expanded_ct, file_suffix, quality_text, debug_df
 
-def add_energy_labels(expanded_ct, df_ei, dic_fuel):
+def add_energy_labels(expanded_ct, df_ei, dic_fuel, dynamic_labels):
     """ Inputs:
     expanded_ct = dataframe of trips that has had Mode_confirm and Replaced_mode added
     dic/df_* = label mappings for energy impact and fuel
     """
     expanded_ct['Mode_confirm_fuel']= expanded_ct['Mode_confirm'].map(dic_fuel)
     expanded_ct = energy_intensity(expanded_ct, df_ei, 'Mode_confirm')
     expanded_ct = energy_footprint_kWH(expanded_ct, 'distance_miles', 'Mode_confirm')
-    expanded_ct = CO2_footprint_lb(expanded_ct, 'distance_miles', 'Mode_confirm')
+
+    if (len(dynamic_labels) > 0):
+        expanded_ct = compute_CO2_footprint_dynamic(expanded_ct, dynamic_labels)
+    else:
+        expanded_ct = CO2_footprint_default(expanded_ct, 'distance_miles', 'Mode_confirm')
     return expanded_ct
 
-def add_energy_impact(expanded_ct, df_ei, dic_fuel):
+def add_energy_impact(expanded_ct, df_ei, dic_fuel, dynamic_labels):
     # Let's first calculate everything for the mode confirm
     # And then calculate everything for the replaced mode
-    expanded_ct = add_energy_labels(expanded_ct, df_ei, dic_fuel)
+    expanded_ct = add_energy_labels(expanded_ct, df_ei, dic_fuel, dynamic_labels)
     expanded_ct['Replaced_mode_fuel']= expanded_ct['Replaced_mode'].map(dic_fuel)
     expanded_ct = energy_intensity(expanded_ct, df_ei, 'Replaced_mode')
     # and then compute the impacts
     expanded_ct = energy_impact_kWH(expanded_ct, 'distance_miles')
-    expanded_ct = CO2_impact_lb(expanded_ct, 'distance_miles')
+
+    if (len(dynamic_labels) > 0):
+        expanded_ct = compute_CO2_impact_dynamic(expanded_ct, dynamic_labels)
+    else:
+        expanded_ct = CO2_impact_default(expanded_ct, 'distance_miles')
```

Comment:
why are we computing CO2 impact in `add_energy_impact`?

Reply:
This is an existing code design.
We call `add_energy_labels()` from `generic_timeseries` notebook and `add_energy_impact()` from `energy_calculations` notebook.

Reply:
the question was more around emissions versus energy. if the function is `add_energy_impact`, why is it calling `CO2_impact`, which is a measure of emissions?

Reply:
I see that we were computing the CO2 footprint earlier; I am still not convinced that this is necessary, but will defer this for now.